extension/storage/storage.go

```diff
@@ -51,6 +53,48 @@ type Client interface {
 	// Delete will delete data associated with the specified key
 	Delete(context.Context, string) error
 
+	// Batch handles specified operations in batch. Get operation results are put in-place
+	Batch(context.Context, ...Operation) error
+
 	// Close will release any resources held by the client
 	Close(context.Context) error
 }
+
+const (
+	Get = iota
+	Set
+	Delete
+)
+
+type operation struct {
+	// Key specifies key which is going to be get/set/deleted
+	Key string
+	// Value specifies value that is going to be set or holds result of get operation
+	Value []byte
+	// Type describes the operation type
+	Type int
+}
+
+type Operation *operation
```

Comment:
This is because for `Get`, the Value is updated in-place. The version with pointers had slightly better throughput actually when I benchmarked it locally

Reply:
> Nit: can the update-in-place be achieved by making just the `Value` field a pointer `*[]byte` while passing Operation by value (and thus reducing GC pressure for Set and Delete cases)?

Perhaps Golang experts could provide more accurate explanation but my understanding is it would still not work due to the way Operation is passed. The update would be only present in the local context (within the function where the update was made). I actually tried this and that was the outcome. 

If the collection of operations would be sent as a slice (rather than variadic function argument) and then the result would be taken from that slice, it would work, but I think also would make things more complex